config/PropertyNames.xml

```diff
@@ -597,6 +597,7 @@ generated from the section label.
         <property name="TruststorePassword" />
         <property name="TruststoreType" />
         <property name="UsePlatformCAs" />
+        <property name="UseMachineContext" />
```

Comment:
Maybe I should just move this as a boolean to the TLS options classes, it is only require to build a custom chain when running without an user profile loaded (local servcie)

Reply:
This is used  in two situations:

- If you define the client/server certificate using IceSSL.CertFile, when importing this certificate by default the key is imported in the user key store, but for services running without user profile loaded the machine key set must be used.
- When building a custom x509 chain because we are using a custom CA, in this case it is used by the chain building process

For importing the private key into the user or machine key set, maybe `IceSSL.UseMatchineKeySet` is a better name.

When building the keychain it is used to load certificates from user or machine context, need to check that we actually need to set this at all, the ExtraStore we populate from the configured CAs should be enough.


 So I considering renaming this property to `IceSSL.UseMatchineKeySet` and use the default constructor for the X509Chain

Reply:
I'm not sure it's a good idea to remove useMachineContext if the user needs to rely on the machine context when not using custom CA certificates. 

Could we simply remove the configuration property and have a boolean on the option class that specifies whether or not to use the machine context/keyset? The bool would be used for two purposes but I don't really see why you'd want to use different values.

I'm thinking it would be simpler if we minimize the number of IceSSL configuration properties that are implementation specific. We could instead privilege specific options on the Tls option classes.

The Windows SChannel C++ impl. will probably similar changes as well.